src/Bicep.Core/Emit/ResourceDependencyVisitor.cs

```diff
@@ -155,6 +161,65 @@ public override void VisitVariableAccessSyntax(VariableAccessSyntax syntax)
             }
         }
 
+        /// <summary>
+        /// Determines whether a reference to a resource is weak.
```

Comment:
>weak [](http://example.com/codeflow?start=60&length=4)

For cases when the index expression is a VariableAccessSyntax, we will return false because we're not sure. 

Should we refect it somewhere in the comments that if this function returns true it's a definitive answer, but false is more of a "maybe"?

Reply:
`false` is a definitive answer, too. If the compiler cannot determine which top-level property of a resource is being accessed, it will generate a reference expression like `reference(<symbolic name>, <api version>, 'Full')[variables(<variable name>)]`. This is a `strong` reference because ARM infers a dependency from the presence of this expression.

Reply:
Looking at the template we generate in this case, the generated expression means that these two templates might return different results:

```bicep
resource r '<type>@<version>' = {
}

output o string = r.name
```

```bicep
param p string = 'name'

resource r '<type>@<version>' = {
}

output o string = r[p]
```

In the first case, the value of `o` will be whatever was supplied for the name of `r`, whereas in the second case, the value of `o` will be the full hierarchical name of the resource. These may be different if `r` specifies a parent instead of using a `/`-delimited name for a child resource.

Definitely out of scope for this PR, but just wanted to note a footgun I hadn't seen before!

Reply:
Makes sense why it's definitely false also. 

The discrepancy in behavior is definitely unexpected! (But not something introduced on this PR.)